### ISSUE TITLE
Getting error ValueError: OPENAI_API_KEY is not set or passed as an argument

### ISSUE THREAD

**Issue by pst2154** [r2d4/openlm#1]
tried with version .5 and .4 on pypi

```
import openlm as openai
import json

completion = openai.Completion.create(
    model="ada", 
    prompt="Hello world",
)
print(json.dumps(completion, indent=4))
```

if I use openai instead it runs perfectly fine

```
import openai as openai

```

**Comment by Nantero1** [comment]
Yes, it seems `cls.register_default()` is called without arguments [here](https://github.com/r2d4/openlm/blob/c2ac78b6e67780457002cb672cc80ea2fd13cb0d/openlm/openlm.py#L55), it should be called with arguments, in case the API key is passed as parameter. Interestingly it is [designed to be able to receive an api key as input parameter](https://github.com/r2d4/openlm/blob/c2ac78b6e67780457002cb672cc80ea2fd13cb0d/openlm/openlm.py#L150C9-L150C25), but is not called in this way.

Here a workaround to silence this issue: 

```python
import openlm

# FIXME: Cannot instantiate the api_key using a parameter (and do not want to reuse OPENAI_API_KEY)
openlm.api_key = os.environ.get("OPEN_LLM_API_KEY")

llm = OpenLM(base_url=...
```
